File: demo/src/app/model/ApiResponseMessage.ts
import { Student } from 'src/app/model/Student';

export class  ApiResponseMessage {
//   message: string;
//   code: string;
//   data: Student[];
   // add other fields that your DTO has

   message : string = '';
    //HttpStatus status;
    code: string ='';
   // boolean success;
   //data: Student[];
   data: object | undefined;
   //private Object data;

}

File: Online-Course-Management-System/src/main/java/com/salman/dto/ApiResponseMessage.java
package com.salman.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.http.HttpStatus;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ApiResponseMessage {

    private String message;
    private HttpStatus status;
    private boolean success;
    private Object data; // This will hold the DTO or any other object
}